Sources/SpeziAccount/AccountValue/Keys/PersonNameKey.swift

```diff
@@ -12,114 +12,94 @@ import SpeziViews
 }
 
 
-// MARK: - UI
-
-extension PersonNameKey {
-    public struct DataDisplay: DataDisplayView {
-        public typealias Key = PersonNameKey
-
-        private let value: PersonNameComponents
+private struct EntryView: DataEntryView {
+    @Environment(Account.self)
+    private var account
 
-        public var body: some View {
-            ListRow(Key.name) {
-                Text(value.formatted(.name(style: .long)))
-            }
-        }
+    @ValidationState private var givenNameValidation
+    @ValidationState private var familyNameValidation
 
+    @Binding private var name: PersonNameComponents
 
-        public init(_ value: PersonNameComponents) {
-            self.value = value
-        }
+    private var nameIsRequired: Bool {
+        account.configuration.name?.requirement == .required
     }
-}
 
-extension PersonNameKey {
-    public struct DataEntry: DataEntryView {
-        public typealias Key = PersonNameKey
+    private var validationRule: ValidationRule {
+        nameIsRequired ? .nonEmpty : .acceptAll
+    }
 
+    var body: some View {
+        Grid(horizontalSpacing: 16) {
+            NameFieldRow(name: $name, for: \.givenName) {
+                Text("UAP_SIGNUP_GIVEN_NAME_TITLE", bundle: .module)
+            } label: {
+                Text("UAP_SIGNUP_GIVEN_NAME_PLACEHOLDER", bundle: .module)
+            }
+                .validate(input: name.givenName ?? "", rules: validationRule)
+                .receiveValidation(in: $givenNameValidation)
+                .focusOnTap() // ensure field receives focus when tapping on the description label
 
-        @Environment(Account.self) private var account
+            GridValidationStateFooter(givenNameValidation.allDisplayedValidationResults)
 
-        @ValidationState private var givenNameValidation
-        @ValidationState private var familyNameValidation
+            Divider()
+                .gridCellUnsizedAxes(.horizontal)
 
-        @Binding private var name: Value
+            NameFieldRow(name: $name, for: \.familyName) {
+                Text("UAP_SIGNUP_FAMILY_NAME_TITLE", bundle: .module)
+            } label: {
+                Text("UAP_SIGNUP_FAMILY_NAME_PLACEHOLDER", bundle: .module)
+            }
+                .validate(input: name.familyName ?? "", rules: validationRule)
+                .receiveValidation(in: $familyNameValidation)
+                .focusOnTap() // ensure field receives focus when tapping on the description label
 
-        private var nameIsRequired: Bool {
-            account.configuration[Key.self]?.requirement == .required
+            GridValidationStateFooter(familyNameValidation.allDisplayedValidationResults)
         }
-        
-        private var validationRule: ValidationRule {
-            nameIsRequired ? .nonEmpty : .acceptAll
+        .environment(\.validationConfiguration, .considerNoInputAsValid)
+        .onChange(of: givenNameValidation) {
+            print("given Name: \(givenNameValidation)")
```

Comment:
Wondering if we want to keep the print statements and others in here?

Reply:
Great catch 👍